api.yaml

```diff
@@ -0,0 +1,448 @@
+          schema:
+            $ref: '#/components/schemas/Error'
+    '403':
+      description: Access forbidden
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    '404':
+      description: Not found
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    '409':
+      description: Duplicate request causing a conflict
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+
+  schemas:
+    Email:
+      description: The email address of the client or the client's contact person
+      nullable: false
+      format: email
+      type: string
+
+    Password:
+      description: The password address of the client or the client's contact person
+      nullable: false
+      format: password
+      type: string
+
+    Uid:
+      description: Идентификатор пользователя в формате UUID
+      type: string
+      example: abcd1234-abcd-1234-abcd-1a2b3c4d
+
+    BookReservationResponse:
+      discriminator:
+        propertyName: status
+        mapping:
+          SUCCESS: '#/components/schemas/BookReservationSuccess'
+          FAILED: '#/components/schemas/BookReservationFailure'
+      oneOf:
+        - $ref: '#/components/schemas/BookReservationSuccess'
+        - $ref: '#/components/schemas/BookReservationFailure'
+
+    BookReservationSuccess:
+      type: object
+      properties:
+        status:
+          type: string
+          enum: [ SUCCESS ]
+        reservationId:
+          type: string
+
+    BookReservationFailure:
+      type: object
+      properties:
+        status:
+          type: string
+          enum: [ FAILED ]
+        reasonCode:
+          type: string
+
+    ReservationRequestResponse:
+      discriminator:
+        propertyName: status
+        mapping:
+          SUCCESS: '#/components/schemas/ReservationRequestSuccess'
+          FAILED: '#/components/schemas/ReservationRequestFailure'
+      oneOf:
+        - $ref: '#/components/schemas/ReservationRequestSuccess'
+        - $ref: '#/components/schemas/ReservationRequestFailure'
+
+    ReservationRequestSuccess:
+      type: object
+      properties:
+        status:
+          type: string
+          enum: [ SUCCESS ]
+        data:
+          type: object
+          properties:
+            id:
+              type: string
+            bookId:
+              type: string
+            userId:
+              type: string
+            reservationStatus:
+              type: string
+
+    ReservationRequestFailure:
+      type: object
+      properties:
+        status:
+          type: string
+          enum: [ FAILED ]
+        reasonCode:
+          type: string
+
+    BookContentResponse:
+      type: object
+      properties:
+        id:
+          type: string
+        title:
+          type: string
+        author:
+          type: string
+        content:
+          type: string
```

Comment:
You include the whole book content? Are we booking service for libraries or an online library?